pkg/http/sender.go

```diff
@@ -0,0 +1,605 @@
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) Int64Column(name string, val int64) *LineSender {
+	s.Buffer.Int64Column(name, val)
+	return s
+}
+
+// Long256Column adds a 256-bit unsigned integer (long256) column
+// value to the ILP message.
+//
+// Only non-negative numbers that fit into 256-bit unsigned integer are
+// supported and any other input value would lead to an error.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) Long256Column(name string, val *big.Int) *LineSender {
+	s.Buffer.Long256Column(name, val)
+	return s
+}
+
+// TimestampColumn adds a timestamp column value to the ILP
+// message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) TimestampColumn(name string, ts time.Time) *LineSender {
+	s.Buffer.TimestampColumn(name, ts)
+	return s
+}
+
+// Float64Column adds a 64-bit float (double) column value to the ILP
+// message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) Float64Column(name string, val float64) *LineSender {
+	s.Buffer.Float64Column(name, val)
+	return s
+}
+
+// StringColumn adds a string column value to the ILP message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) StringColumn(name, val string) *LineSender {
+	s.Buffer.StringColumn(name, val)
+	return s
+}
+
+// BoolColumn adds a boolean column value to the ILP message.
+//
+// Column name cannot contain any of the following characters:
+// '\n', '\r', '?', '.', ',', ”', '"', '\', '/', ':', ')', '(', '+',
+// '-', '*' '%%', '~', or a non-printable char.
+func (s *LineSender) BoolColumn(name string, val bool) *LineSender {
+	s.Buffer.BoolColumn(name, val)
+	return s
+}
+
+// Close closes the underlying HTTP client. If no clients remain open,
+// the global http.Transport will close all idle connections.
+func (s *LineSender) Close() {
+	newCt := clientCt.Add(-1)
+	if newCt == 0 {
+		globalTransport.CloseIdleConnections()
+	}
+}
+
+// AtNow omits the timestamp and finalizes the ILP message.
+// The server will insert each message using the system clock
+// as the row timestamp.
+//
+// If the underlying buffer reaches configured capacity, this
+// method also sends the accumulated messages.
+func (s *LineSender) AtNow(ctx context.Context) error {
+	err := s.Buffer.At(time.Time{}, false)
+	if err != nil {
+		return err
+	}
+	if s.Len() > s.BufCap {
```

Comment:
Other ILP/HTTP clients don't have this logic. It can be met only in ILP/TCP clients. For HTTP we flush either on `Flush` or when `auto_flush_rows` is reached. The same applies to `At`.

Reply:
What is the reason for that? I can easily remove the code (and the `WithBufferCapacity` http sender option too), I'm just curious

Reply:
That's because in ILP/HTTP we want the user to have full control over transaction boundaries, i.e. by default commit(s) should only happen on `Flush`. Then the `auto_flush_rows` setting is exposed for those users who don't care about transactions and just let the client flush occasionally.